collect_app/src/main/java/org/odk/collect/android/widgets/GeoPointMapWidget.java

```diff
@@ -0,0 +1,210 @@
+        binding.geoAnswerText.setText(getAnswerToDisplay(stringAnswer));
+
+        if (binding.geoAnswerText.getText().toString().equals("")) {
+            stringAnswer = "";
+        }
+
+        updateButtonLabelsAndVisibility(stringAnswer != null);
+        widgetValueChanged();
+    }
+
+    private void determineMapProperties(QuestionDef questionDef) {
+        // Determine the accuracy threshold to use.
+        String acc = questionDef.getAdditionalAttribute(null, ACCURACY_THRESHOLD);
+        accuracyThreshold = acc != null && !acc.isEmpty() ? Double.parseDouble(acc) : DEFAULT_LOCATION_ACCURACY;
+
+        // Determine whether the point should be draggable.
+        if (hasAppearance(getFormEntryPrompt(), PLACEMENT_MAP)) {
+            draggable = true;
+        } else if (hasAppearance(getFormEntryPrompt(), MAPS)) {
+            draggable = false;
+        }
+    }
+
+    private String getAnswerToDisplay(String answer) {
+        try {
+            if (answer != null && !answer.isEmpty()) {
+                String[] parts = answer.split(" ");
+                return getContext().getString(
+                        R.string.gps_result,
+                        GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[0]), "lat"),
+                        GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[1]), "lon"),
+                        GeoWidgetUtils.truncateDouble(parts[2]),
+                        GeoWidgetUtils.truncateDouble(parts[3])
+                );
+            }
+        } catch (NumberFormatException e) {
+            return "";
+        }
+        return "";
+    }
+
+    private void updateButtonLabelsAndVisibility(boolean dataAvailable) {
+        if (readOnly) {
+            binding.simpleButton.setText(R.string.geopoint_view_read_only);
+        } else {
+            binding.simpleButton.setText(
+                    dataAvailable ? R.string.view_change_location : R.string.get_point);
+        }
+    }
+
+    private void onButtonClick() {
+        if (MultiClickGuard.allowClick(QuestionWidget.class.getName())) {
+            getPermissionUtils().requestLocationPermissions((Activity) getContext(), new PermissionListener() {
+                @Override
+                public void granted() {
+                    waitingForDataRegistry.waitForData(getFormEntryPrompt().getIndex());
+                    startGeoActivity();
+                }
+
+                @Override
+                public void denied() {
+                }
+            });
+        }
+    }
+
+    private void startGeoActivity() {
+        Context context = getContext();
+        Intent intent = new Intent(context, GeoPointMapActivity.class);
+
+        if (stringAnswer != null && !stringAnswer.isEmpty()) {
+            intent.putExtra(LOCATION, GeoWidgetUtils.getLocationParamsFromStringAnswer(stringAnswer));
+        }
+        intent.putExtra(READ_ONLY, readOnly);
```

Comment:
Note here that the `GeoPointMapActivity` accepts an intent extra to represent whether the map should be readonly. So it's very clear that the intended behavior was to show a readonly map.